src/widget/wfindonmenu.cpp

```diff
@@ -0,0 +1,279 @@
+#include "widget/wfindonmenu.h"
+
+#include <QDesktopServices>
+#include <QScreen>
+#include <QUrl>
+#include <QUrlQuery>
+#include <QtDebug>
+
+#include "track/track.h"
+#include "util/math.h"
+#include "util/qt.h"
+#include "util/widgethelper.h"
+
+namespace {
+
+QString composeActionText(const QString& prefix, const QString& trackProperty) {
+    return prefix + QStringLiteral(" | ") + trackProperty;
+}
+
+QString composeSearchQuery(const QString& trackAlbumOrTitle, const QString& artistTitle) {
+    return trackAlbumOrTitle + QStringLiteral(" ") + artistTitle;
+}
+
+QString composePrefixAction(WFindOnMenu::TrackProperty trackProperty) {
```

Comment:
Are these supposed to be i18n strings that need to be translated? Then this won't work.

Reply:
I think this needs to be translated. In order to make them i18n strings, According to QT Doc: [Writing Source Code for Translation](https://doc.qt.io/qt-5/i18n-source-translation.html) I should return QString instead of QStringLiteral, right?